### PR TITLE
OSAC-4488: Address CaaS test review feedback from PR #490

### PR DESCRIPTION
**Pull request by tzvatot**
## [OSAC-4488](https://redhat.atlassian.net/browse/OSAC-4488): Address caas test review feedback from PR #490

**Jira:** https://redhat.atlassian.net/browse/OSAC-4488
**Parent epic:** [OSAC-3593](https://redhat.atlassian.net/browse/OSAC-3593) (migrate e2e test suite into the osac mono-repo)
**Follows:** #490 (caas e2e migration)

### Summary

Addresses the automated-review findings raised on the merged caas e2e migration
(#490) so the migrated `tests/e2e/caas/` suite is correct, secure, and
consistent with repo conventions. Test-only change — no product code.

### Changes

**`tests/e2e/caas/conftest.py`**
- Quote `namespace` and `service_account` with `shlex.quote()` in the
  `oc create token` command passed to `osac login --token-script` (the CLI runs
  that string in a shell) — closes a shell-injection vector.
- Add docstrings to the `cli`, `cluster_template`, `pull_secret_path`, and
  `ssh_public_key_path` fixtures.

**`tests/e2e/caas/test_cluster_create.py`**
- Assert `node_sets` is non-empty before deriving the component count and
  scaling a worker set, so an empty spec surfaces a clear failure instead of an
  opaque `StopIteration`.
- Clean up the `ClusterVersion` resources created by
  `test_cluster_create_with_version` and
  `test_cluster_create_rejected_for_invalid_version` in `finally` blocks so
  repeated runs don't leave stale versions on the shared cluster. Cleanup is
  robust on failure paths: resources are created inside the `try`, versions are
  tracked as they are created, and a referenced version's delete waits for full
  cluster removal first (instant on the happy path).
- Add a docstring to `test_cluster_create`.

### Testing

- **Unit tests:** N/A (e2e suite).
- **Integration/e2e tests:** the suite requires a live cluster and runs in CI
  (`e2e-caas-full-install`). Behavioral changes here are an in-test assertion and
  teardown hardening; no new test scenario is introduced.
- **Local validation:** `ruff check` clean, `ruff format --check` clean,
  `pytest tests/e2e/caas/ --collect-only` collects all 4 tests, and secret scans
  (`gitleaks`, `detect-private-key`) pass.

### Review findings addressed (from #490)

- [x] StopIteration edge case + assert `node_sets` non-empty
- [x] Missing ClusterVersion cleanup on failure paths
- [x] Shell injection in `conftest.py` token-script
- [x] Sensitive data in logs (verified already mitigated — token never enters
      Python; cleanup suppresses without logging)
- [x] Docstring coverage
- [x] AI attribution — `Assisted-by:` trailers, no `Co-Authored-By`

## Summary

- **Tests:** Updated the migrated `tests/e2e/caas/` E2E suite.
- **Security:** Added shell quoting for namespace and service-account values in the `oc create token` command.
- **Test reliability:** Added fixture and test docstrings, validated non-empty `node_sets`, and improved cleanup for clusters and `ClusterVersion` resources.
- **Resource cleanup:** Failed cluster-version tests now wait for `ClusterOrder` deletion before waiting for gRPC removal.
- **Product code:** No API, controller, database, authentication, deployment, or production-code changes.
- **Backward compatibility:** No impact is expected.

## Risk classification

**risk:ship** — The change is limited to test fixtures and E2E tests. It does not change production behavior or public interfaces. Lint, formatting, test collection, and secret-scan checks passed.

The change is not **risk:show** because it has no production behavior impact. It is not **risk:ask** because it introduces no unresolved production risk.

### CODE REVIEW

File: tests/e2e/caas/test_cluster_create.py [hunk 1]
```diff
@@ -201,36 +208,55 @@ def test_cluster_create_with_version(
...
     finally:
-        with contextlib.suppress(subprocess.CalledProcessError):
-            cli.delete_cluster(uuid=uuid)
+        # On the failure path the cluster may still exist and reference the
+        # version; a referenced ClusterVersion cannot be deleted, so remove the
+        # cluster and wait for it to be gone first. Skipped entirely on the happy
+        # path, where the body already deleted the cluster and cleared uuid.
+        if uuid is not None:
+            with contextlib.suppress(subprocess.CalledProcessError):
+                cli.delete_cluster(uuid=uuid)
```
[the remark this answers is not in the document]
Comment (tzvatot) [reply tests/e2e/caas/test_cluster_create.py]: Suppressing `SubprocessError` (the common base of `CalledProcessError` and `TimeoutExpired`) instead of just `CalledProcessError` covers both cases. Fixed in 990868c.

File: tests/e2e/caas/test_cluster_create.py [hunk 2]
```diff
@@ -201,36 +208,55 @@ def test_cluster_create_with_version(
...
+            with contextlib.suppress(TimeoutError):
+                wait_for_cluster_grpc_removal(grpc=grpc, uuid=uuid)
+        private_grpc.call_unchecked(service="osac.private.v1.ClusterVersions/Delete", data={"id": version["id"]})
```
[the remark this answers is not in the document]
Comment (tzvatot) [reply tests/e2e/caas/test_cluster_create.py]: Intentionally best-effort - adding assertions in finally cleanup would mask the actual test failure if both fail simultaneously. `ensure_cluster_version` is idempotent by name, so leaked versions don't accumulate across runs.

**Review by sk-ilya — APPROVED** [review]
